src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -131,7 +131,7 @@ public static string GetFullPath(string path, string basePath)
             // them properly. As such we need to manually remove segments and not use GetFullPath().
 
             return PathInternal.IsDevice(combinedPath.AsSpan())
-                ? PathInternal.RemoveRelativeSegments(combinedPath, PathInternal.GetRootLength(combinedPath.AsSpan()))
+                ? PathInternal.RemoveRelativeSegments(combinedPath)
```

Comment:
Rename all instances of RemoveRelativeSegments to RemoveRedundantSegments for consistency?